src/server/game/Entities/GameObject/GameObject.cpp

```diff
@@ -419,6 +419,11 @@ void GameObject::Update(uint32 diff)
                     m_SkillupList.clear();
                     m_usetimes = 0;
 
+                    // If nearby linked trap exists, respawn it
+                    if (uint32 linkedEntry = GetGOInfo()->GetLinkedGameObjectEntry())
+                        if (GameObject* linkedTrap = FindNearestGameObject(linkedEntry, 10.0f))
```

Comment:
Suggestion: instead of using a generic "shotgun" grid searcher, why don't save spawned linked GO spawn GUID in the GameObject, then just do a ObjectAccessor:GetGameObject(*this, linkedGoGUID) to despawn only the linked one.